test/e2e/util.go

```diff
@@ -454,6 +448,91 @@ func newRequestFromForm(form *html.Node, currentURL *url.URL, user string) (*htt
 	return req, nil
 }
 
+// Varying the login name for each test ensures we test a fresh grant
+func createTestIdP(
+	t *testing.T,
+	kubeClient *kubernetes.Clientset,
+	oauthClient configv1client.OAuthInterface,
+	userClientSet *userclients.Clientset,
+	nsName string,
+	numUsers int,
+) ([]string, func()) {
+	oauthConfig, err := oauthClient.Get(context.TODO(), "cluster", metav1.GetOptions{})
+	require.NoError(t, err)
+
+	var users []string
+	for i := 0; i < numUsers; i++ {
+		users = append(users, fmt.Sprintf("testuser%d", i))
+	}
+
+	kubeClient.CoreV1().Secrets("openshift-config").Create(context.TODO(),
+		&corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: nsName + "htpasswd",
```

Comment:
nit: `nsName + "htpasswd"` is used multiple times. maybe use a varaible?

Reply:
done